dev-cpp/sourcetrail/sourcetrail-2018.3.55.ebuild

```diff
@@ -17,7 +17,7 @@ IUSE="examples selinux"
 DEPEND="dev-util/patchelf"
 
 RDEPEND="
-	dev-libs/openssl
+	=dev-libs/openssl-1.0*:*
```

Comment:
* `:*` slot operator is contradicting your version restriction
* Instead of restricting to`1.0`, you should depend on the correct binary openssl slot
* A change of `RDEPEND` requires revbump.

Reply:
Thanks for the feedback :)

I have adjusted the dependency to dev-libs/openssl:0 - I hope that is correct? I am not entirely sure how OpenSSL handles binary backwards compatibility, but I assume that any version within slot 0 is binary compatible? As I noted in the commit message, I can see from the binaries that the Sourcetrail package has been built against OpenSSL 1.0.1f.

Also, I haven't done a revbump before, so please let me know if I haven't done it correctly :)

Reply:
slot 0 is for source packages, while 1.0* versions would currently work for your package, slot 0 is already providing 1.1.0 and 1.1.1 as well. so if your binary package needs 1.0, depend on slot `1.0.0`.

Reply:
So would that mean that the ebuild should depend on:

`=dev-libs/openssl:1.0.0`

or would it be

`=dev-libs/openssl-1.0.0*`

or how would you express that dependency?

Reply:
For depending on the slot, you use: `dev-libs/openssl:1.0.0`

Reply:
But now I get a huge list of blocked packages, it appears to be caused by dev-libs/openssl:1.0.0 and dev-libs/openssl:0 cannot be installed at the same time :/ This is far from optimal :) So what can I do to fix this?
If I do a `grep -r 'dev-libs/openssl:[1-9]+' /usr/portage` I don't see any mathes, so I am wondering what other binary packages are doing?
Looking at `app-emulation/crossover-bin-17.5.1.ebuild` the OpenSSL dependency is listed as `dev-libs/openssl:0[abi_x86_32(-)]` and for `media-sound/spotify` it is simply `dev-libs/openssl:0`

Any help is greatly appreciated

This is an extract of the blockers I see:
```
/usr/local/portage/dev-cpp/sourcetrail emerge -1 sourcetrail                       

These are the packages that would be merged, in order:

Calculating dependencies       ... done!                 
[ebuild  NS   ~] dev-libs/openssl-1.0.2q-r200:1.0.0::gentoo [1.0.2q:0::gentoo] USE="asm sslv3 tls-heartbeat zlib -bindist -gmp -kerberos -rfc3779 -sctp -sslv2 -static-libs -test -vanilla" ABI_X86="(64) -32 (-x32)" CPU_FLAGS_X86="(sse2)" 0 KiB
[ebuild   R   ~] dev-cpp/sourcetrail-2018.3.55-r1::DavidsOverlay  USE="-examples (-selinux)" 0 KiB
[blocks B      ] =dev-libs/openssl-1.0.2*:0 ("=dev-libs/openssl-1.0.2*:0" is blocking dev-libs/openssl-1.0.2q-r200)

Total: 2 packages (1 in new slot, 1 reinstall), Size of downloads: 0 KiB
Conflict: 1 block (1 unsatisfied)

 * Error: The above package list contains packages which cannot be
 * installed at the same time on the same system.

  (dev-libs/openssl-1.0.2q-r200:1.0.0/1.0.0::gentoo, ebuild scheduled for merge) pulled in by
    dev-libs/openssl:1.0.0 required by (dev-cpp/sourcetrail-2018.3.55-r1:0/0::DavidsOverlay, ebuild scheduled for merge)
    >=dev-libs/openssl-0.9.8:* required by (net-vpn/openvpn-2.4.6:0/0::gentoo, installed)

  (dev-libs/openssl-1.0.2q:0/0::gentoo, installed) pulled in by
    dev-libs/openssl:0/0= required by (dev-db/mariadb-10.1.37:0/18::gentoo, installed)
    dev-libs/openssl:0= required by (www-client/falkon-3.0.1:0/0::gentoo, installed)
    dev-libs/openssl:0/0= required by (dev-libs/libgit2-0.26.8:0/26::gentoo, installed)
    >=dev-libs/openssl-0.9.8:0= required by (dev-python/m2crypto-0.24.0:0/0::gentoo, installed)
```

Reply:
> But now I get a huge list of blocked packages, it appears to be caused by dev-libs/openssl:1.0.0 and dev-libs/openssl:0 cannot be installed at the same time :/ This is far from optimal :) So what can I do to fix this?

When you look closer it is not the slots that are conflicting, but of course `dev-libs/openssl-1.0.2q-r200:1.0.0` needs to block `=dev-libs/openssl-1.0.2*:0` which would install the same files.
The blocker happens because you only make a partial upgrade, and you are apparently coming from a stable system. But sourcetrail only has `~arch` keywords, so it is perfectly fine to be blocking `arch` versions.